### PR TITLE
Added the ability to close file handles

### PR DESCRIPTION
**Pull request by malexdev**
I added the ability to close file handles via new `ReadCloser` and `ScanCloser` interfaces.

### CODE REVIEW

File: utfutil.go [hunk 1]
```diff
@@ -41,27 +41,112 @@ import (
...
+	Text() string
+	Close() error
+}
+
+type scanCloser struct {
+	file    UTFReadCloser
+	scanner *bufio.Scanner
+}
+
+// Buffer will run the Buffer functoin on the underlying bufio.Scanner.
```
Comment (TomOnTime) [inline utfutil.go]: Please s/functoin/function/g

File: README.md [hunk 2]
```diff
@@ -2,6 +2,10 @@
 
 Utilities to make it easier to read text encoded as UTF-16.
 
+## Forked
```
Comment (TomOnTime) [inline README.md]: This should be removed when we merge.

**Comment by TomOnTime** [comment]
Wow!  I had no idea anyone was using this module besides me.  I'm super excited to accept this PR. I had a few cosmetic suggestions.

**Comment by malexdev** [comment]
@TomOnTime Yes, this module has been a life saver! Thanks so much for making it! :)

**Comment by TomOnTime** [comment]
Thanks for the PR!  You really cleaned up my mess! ;-)

**Pull request merged**